### DISCUSSION TITLE
Assign errors for all fields

### DISCUSSION THREAD

**Discussion by D3strukt0r** [react-hook-form/react-hook-form#11820] [Q&A] [answered]
Let's say, i don't want to duplicate a complex (or needs other data only available in the backend) validation in the frontend and backend, so i just have it in the backend. The backend now creates an object/array of all the fields with their errors. Is there a way to tell this library, "hey here are all the errors, please make them available in 'formState.errors'"?

**Comment by nicostombros** [comment, marked as answer]
Do you mean with something like `setError`?
https://react-hook-form.com/docs/useform/seterror
You probably want to loop over each of your errors and do setError to map the field to the error

**Reply by D3strukt0r** [reply]
Yes, thank you. why is this so hidden 🙈 To me practically a must have feature.
I had to map my Symfony backend errors to the frontend, here is my helper function if anyone is interested https://github.com/D3strukt0r/wedding-manuele-robine/blob/develop/pwa/src/utils/form.ts and here how i'd use it https://github.com/D3strukt0r/wedding-manuele-robine/blob/9dab7fe7976afe326ab2ae0a7fee8cb4b1acb139/pwa/src/pages/admin/users/ListUsers.tsx#L64
Typescript is not my first priority, so they aren't perfect though
